app/src/main/java/ch/sdp/vibester/activity/ProfileActivity.kt

```diff
@@ -41,41 +48,65 @@ class ProfileActivity : AppCompatActivity() {
 
         setContentView(R.layout.activity_profile)
 
-        val editUsername = findViewById<Button>(R.id.editUser)
+        setEditUserNameBtnListener()
+        setLogOutBtnListener()
+        setRetToMainBtnListener()
+        setShowQrCodeBtnListener()
+        setQrCodeToProfileBtnListener()
+
+        // Do not enable querying database while executing unit test
+        val isUnitTest: Boolean = intent.getBooleanExtra("isUnitTest", false)
 
-        val logoutButton = findViewById<Button>(R.id.logout)
+        if (!isUnitTest) {
+            queryDatabase()
+        } else {
+            val upTest: User? = intent.getSerializableExtra("userTestProfile") as User?
+            if (upTest == null) {
+                setupProfile(User())
+            } else {
+                setupProfile(upTest)
+            }
+        }
 
-        val retToMain = findViewById<FloatingActionButton>(R.id.profile_returnToMain)
+    }
 
-        editUsername.setOnClickListener {
+    private fun setEditUserNameBtnListener() {
+        findViewById<Button>(R.id.editUser).setOnClickListener {
             showGeneralDialog(R.id.username, "username")
         }
+    }
 
-        retToMain.setOnClickListener{
+    private fun setLogOutBtnListener() {
+        findViewById<Button>(R.id.logout).setOnClickListener {
+            FirebaseAuth.getInstance().signOut()
             IntentSwitcher.switchBackToWelcome(this)
             finish()
         }
+    }
 
-        logoutButton.setOnClickListener{
-            FirebaseAuth.getInstance().signOut()
+    private fun setRetToMainBtnListener() {
+        findViewById<FloatingActionButton>(R.id.profile_returnToMain).setOnClickListener {
             IntentSwitcher.switchBackToWelcome(this)
             finish()
         }
+    }
 
-        // Do not enable querying database while executing unit test
-        val isUnitTest: Boolean = intent.getBooleanExtra("isUnitTest", false)
+    private fun setShowQrCodeBtnListener() {
+        findViewById<Button>(R.id.showQRCode).setOnClickListener {
+            setLayoutVisibility(R.id.QrCodePage, true)
+            setLayoutVisibility(R.id.profileContent, false)
+        }
+    }
 
-        if (!isUnitTest) {
-            queryDatabase()
-        } else {
-            var upTest: User? = intent.getSerializableExtra("userTestProfile") as User?
-            if (upTest == null) {
-                setupProfile(User())
-            } else {
-                setupProfile(upTest)
-            }
+    private fun setQrCodeToProfileBtnListener() {
+        findViewById<FloatingActionButton>(R.id.qrCode_returnToProfile).setOnClickListener {
+            setLayoutVisibility(R.id.QrCodePage, false)
+            setLayoutVisibility(R.id.profileContent, true)

```

Comment:
```suggestion
        findViewById<FloatingActionButton>(R.id.qrCode_returnToProfile).setOnClickListener {
        //Otherwise not able to generate qr
        if(user.uid != ""){
            setLayoutVisibility(R.id.QrCodePage, false)
            setLayoutVisibility(R.id.profileContent, true)
            }

```

Reply:
This check is not necessary because after login every user will have a uid which is not null